src/routes/console/project-[project]/storage/bucket-[bucket]/settings/+page.svelte

```diff
@@ -69,78 +69,72 @@
         } else isExtensionsDisabled = true;
     }
 
-    async function toggleBucket() {
-        try {
-            await sdkForProject.storage.updateBucket(
-                $bucket.$id,
-                $bucket.name,
-                undefined,
-                undefined,
-                enabled
-            );
-            invalidate(Dependencies.BUCKET);
-            addNotification({
-                message: `${$bucket.name} has been updated`,
-                type: 'success'
-            });
-            trackEvent('submit_bucket_enable', {
-                value: enabled
-            });
-        } catch (error) {
-            addNotification({
-                message: error.message,
-                type: 'error'
-            });
-        }
+    interface IUpdateBucket {
+        id: string;
+        name: string;
+        permissions?: string[];
+        fileSecurity?: boolean;
+        enabled?: boolean;
+        maximumFileSize?: number;
+        allowedFileExtensions?: string[];
+        compression?: string;
+        encryption?: boolean;
+        antivirus?: boolean;
     }
-    async function updateName() {
-        try {
-            await sdkForProject.storage.updateBucket($bucket.$id, bucketName);
-            invalidate(Dependencies.BUCKET);
-            addNotification({
-                message: 'Name has been updated',
-                type: 'success'
-            });
-            trackEvent('submit_bucket_update_name');
-        } catch (error) {
-            addNotification({
-                message: error.message,
-                type: 'error'
-            });
-        }
+
+    interface IUpdateBucketMisc {
+        successMessage?: string;
+        trackEventName: string;
+        trackEventData?: { value: boolean };
+        arePermsDisabled?: boolean;
     }
-    async function updatePermissions() {
-        try {
-            await sdkForProject.storage.updateBucket($bucket.$id, $bucket.name, bucketPermissions);
-            invalidate(Dependencies.BUCKET);
-            arePermsDisabled = true;
-            addNotification({
-                message: 'Permissions have been updated',
-                type: 'success'
-            });
-            trackEvent('submit_bucket_update_permissions');
-        } catch (error) {
-            addNotification({
-                message: error.message,
-                type: 'error'
-            });
-        }
+
+    function getDefaults(): IUpdateBucket {
+        return {
+            id: $bucket.$id,
+            name: $bucket.name,
+            permissions: $bucket.$permissions,
+            fileSecurity: $bucket.fileSecurity,
+            enabled: $bucket.enabled,
+            maximumFileSize: $bucket.maximumFileSize,
+            allowedFileExtensions: $bucket.allowedFileExtensions,
+            compression: $bucket.compression,
+            encryption: $bucket.encryption,
+            antivirus: $bucket.antivirus
+        };
     }
```

Comment:
I'm not sure if this is necessary...can't we just spread `$bucket` instead?

Reply:
Done